crates/nu-command/src/filesystem/util.rs

```diff
@@ -182,11 +165,19 @@ pub mod users {
         // The capacity for `*groups` is passed in as `*ngroups` which is the buffer max length/capacity (as we initialize with 0)
         // Following reads from `*groups`/`buff` will only happen after `buff.truncate(*ngroups)`
         #[cfg(target_os = "macos")]
-        let res =
-            unsafe { libc::getgrouplist(name.as_ptr(), gid as i32, buff.as_mut_ptr(), &mut count) };
+        let res = unsafe {
+            nix::libc::getgrouplist(
+                name.as_ptr(),
+                gid.as_raw() as i32,
+                buff.as_mut_ptr(),
+                &mut count,
+            )
+        };
```

Comment:
The `nix` docs say that on macOS we may want to do something different. https://docs.rs/nix/latest/src/nix/unistd.rs.html#1664-1666

But we can keep this for the sake of this refactor PR